compiler/src/iree/compiler/Dialect/Util/IR/UtilOps.td

```diff
@@ -579,9 +579,124 @@ def Util_InitializerOp : Util_Op<"initializer", [
   let hasCanonicalizer = 1;
 }
 
+def Util_FuncOp : Util_Op<"func", [
+  AffineScope,
+  AutomaticAllocationScope,
+  FunctionOpInterface,
+  IsolatedFromAbove,
+  OpAsmOpInterface,
+]> {
+  let summary = [{function operation containing a CFG region}];
+  let description = [{
+    An operation declaring a function and optionally defining its contents.
+  }];
```

Comment:
Are there any other comments from https://github.com/llvm/llvm-project/blob/184ca39529a93e69eb175861d7fff5fc79988e53/mlir/include/mlir/Dialect/Func/IR/FuncOps.td#L222-L335 that you'd like to include here?

More context on why contents are optional could be useful, and that has this text:

> An external function declaration (used when
    referring to a function declared in some other module) has no body.